client/src/app/AppParent.js

```diff
@@ -259,6 +267,39 @@ export default class AppParent extends PureComponent {
     });
   }
 
+  /**
+   *
+   * @param {string} category
+   * @param {Error|{ message: string }} errorLike
+   * @param {Tab} tab
+   */
+  logToClient(category, errorLike, tab) {
+    const entry = this.getLogEntry(category, errorLike.message, tab);
+
+    this.triggerAction(null, 'log', entry);
+  }
+
+  getLogEntry(category, message, tab) {
+    if (tab) {
+      message = `[${tab.file.name}] ${message}`;
```

Comment:
Why don't we log the full file path, if it is available to us? It can happen that files are named exactly the same but have very different contents.